src/Agent.Worker/Build/BuildCommandExtension.cs

```diff
@@ -13,43 +14,28 @@
 
 namespace Microsoft.VisualStudio.Services.Agent.Worker.Build
 {
-    public sealed class BuildCommandExtension : AgentService, IWorkerCommandExtension
+    public sealed class BuildCommandExtension: BaseWorkerCommandExtension
     {
-        public Type ExtensionType => typeof(IWorkerCommandExtension);
-
-        public string CommandArea => "build";
-
-        public HostTypes SupportedHostTypes => HostTypes.All;
-
-        public void ProcessCommand(IExecutionContext context, Command command)
+        public BuildCommandExtension()
         {
-            if (string.Equals(command.Event, WellKnownBuildCommand.UploadLog, StringComparison.OrdinalIgnoreCase))
-            {
-                ProcessBuildUploadLogCommand(context, command.Data);
-            }
-            else if (string.Equals(command.Event, WellKnownBuildCommand.UploadSummary, StringComparison.OrdinalIgnoreCase))
-            {
-                ProcessBuildUploadSummaryCommand(context, command.Data);
-            }
-            else if (string.Equals(command.Event, WellKnownBuildCommand.UpdateBuildNumber, StringComparison.OrdinalIgnoreCase))
-            {
-                ProcessBuildUpdateBuildNumberCommand(context, command.Data);
-            }
-            else if (string.Equals(command.Event, WellKnownBuildCommand.AddBuildTag, StringComparison.OrdinalIgnoreCase))
-            {
-                ProcessBuildAddBuildTagCommand(context, command.Data);
-            }
-            else
-            {
-                throw new Exception(StringUtil.Loc("BuildCommandNotFound", command.Event));
-            }
+            CommandArea = "build";
+            SupportedHostTypes = HostTypes.All;
+            InstallWorkerCommand(new ProcessBuildUploadLogCommand());
+            InstallWorkerCommand(new ProcessBuildUploadSummaryCommand());
+            InstallWorkerCommand(new ProcessBuildUpdateBuildNumberCommand());
+            InstallWorkerCommand(new ProcessBuildAddBuildTagCommand());
         }
+    }
 
-        private void ProcessBuildUploadLogCommand(IExecutionContext context, string data)
+    public class ProcessBuildUploadLogCommand: IWorkerCommand
+    {
+        public string Name => "uploadlog";
```

Comment:
Is this the same as what was before WellKnownBuildCommand.*?

Reply:
it should be